### DISCUSSION TITLE
ResBlock Kernel Size

### DISCUSSION THREAD

**Discussion by BK0x7C8** [Project-MONAI/tutorials#1524] [Q&A] [answered]
Hi,

I'd like to build a customized SegResNet, where the blocks have different kernel sizes (according to a paper that I found, which handles feature extraction in a similar case as mine with these special kernel sizes).

However, if I construct a network using ResBlocks, I cannot change the kernel sizes, although there is an arg for that:

`resblock = monai.networks.blocks.ResBlock(
            spatial_dims=3, 
            in_channels=8, 
            norm=('GROUP', {'num_groups': 4}), 
            kernel_size=5, 
            act=('RELU', {'inplace': True})
            )
`
Output:

ResBlock(
  (norm1): GroupNorm(4, 8, eps=1e-05, affine=True)
  (norm2): GroupNorm(4, 8, eps=1e-05, affine=True)
  (act): ReLU(inplace=True)
  (conv1): Convolution(
    (conv): Conv3d(8, 8, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
  )
  (conv2): Convolution(
    (conv): Conv3d(8, 8, kernel_size=(3, 3, 3), stride=(1, 1, 1), padding=(1, 1, 1), bias=False)
  )
)

Note, that although I passed a kernel_size = 5 arg, they are still size 3. Any hints? Thanks!

**Comment by KumoLiu** [comment, marked as answer]
Hi @r00tdotexe, thanks for pointing it out.
Looks like a bug, `kernel_size` didn't correctly send to the convolution.
https://github.com/Project-MONAI/MONAI/blob/281cb0119c01eaa8e6c841880b91f92f45e8d7f7/monai/networks/blocks/segresnet_block.py#L76

**Reply by BK0x7C8** [reply]
Hi @KumoLiu ,

thanks. I took the source code of the ResBlock, (the one you sent), and it seems, that the kernel_size arg is not passed to the get_convl_layer call, in the resblock class (last line in code below). I added it manually in my IDE and now it works!

```
class ResBlock(nn.Module):
    def __init__(
        self,
        spatial_dims: int,
        in_channels: int,
        norm: tuple | str,
        kernel_size: int = 3,
        act: tuple | str = ("RELU", {"inplace": True}),
    ) -> None:
        """
        Args:
            spatial_dims: number of spatial dimensions, could be 1, 2 or 3.
            in_channels: number of input channels.
            norm: feature normalization type and arguments.
            kernel_size: convolution kernel size, the value should be an odd number. Defaults to 3.
            act: activation type and arguments. Defaults to ``RELU``.
        """

        super().__init__()

        if kernel_size % 2 != 1:
            raise AssertionError("kernel_size should be an odd number.")

        self.norm1 = get_norm_layer(name=norm, spatial_dims=spatial_dims, channels=in_channels)
        self.norm2 = get_norm_layer(name=norm, spatial_dims=spatial_dims, channels=in_channels)
        self.act = get_act_layer(act)
        self.conv1 = get_conv_layer(spatial_dims, in_channels=in_channels, out_channels=in_channels, kernel_size=kernel_size)
        self.conv2 = get_conv_layer(spatial_dims, in_channels=in_channels, out_channels=in_channels, kernel_size=kernel_size)

```
